src/main/java/com/softserveinc/ps4j/challenge/round001/CostOfLinkingTrainsProblem.java

```diff
@@ -8,7 +11,44 @@
 class CostOfLinkingTrainsProblem {
 
     int solve(int[] trains) {
-        throw new UnsupportedOperationException("not yet implemented");
-    }
 
+        if (trains.length == 0) {
+            return 0;
+        }
+        if (trains.length == 1) {
+            return trains[0];
+        }
+        if (trains.length == 2) {
+            return IntStream.of(trains).sum();
+        }
+
+        var cost = 0;
+        var trainsList = new ArrayList<Integer>();
```

Comment:
This does work, but I suggest exploring other, more efficient solutions.

Reply:
Do you mean the whole implementation or using ArrayList?

Reply:
Refactored.